### ISSUE TITLE
Error : Weights File

### ISSUE THREAD

**Issue by divijkhaitan** [google-deepmind/graphcast#147]
The paper says that the embedder for the mesh nodes takes in 3 input features. But there isn't a single matrix that would correspond to the weights for the mesh node embedder, since the mesh node encoder also has a input dimension of 474 for the weight matrix ('params:grid2mesh_gnn/~_networks_builder/encoder_nodes_mesh_nodes_mlp/~/linear_0:w'). Is this meant to have a dimension of 474?

**Comment by andrewlkd** [comment]
Hey,

Just to check, is there an actual error occurring on your end? If so, please include some stack traces and/or error messages.

Otherwise, note that we [pad](https://github.com/google-deepmind/graphcast/blob/7077d40a36db6541e3ed72ccaed1c0d202fa6014/graphcast/graphcast.py#L632) the mesh node input features, hence why the shape may not align with what you are expecting.

Best,

Andrew

**Comment by divijkhaitan** [comment]
I'm attempting to analyse the weights of the model, there isn't an error when running the code in this regard. Would I be correct to undo this padding by keeping the first three rows and removing the remaining 471 to get the weights + bias corresponding to the 3 input features?

**Comment by andrewlkd** [comment]
I see. 

As per the code

```
    dummy_mesh_node_features = jnp.zeros(
        (self._num_mesh_nodes,) + grid_node_features.shape[1:],
        dtype=grid_node_features.dtype)
    new_mesh_nodes = mesh_nodes._replace(
        features=jnp.concatenate([
            dummy_mesh_node_features,
            _add_batch_second_axis(
                mesh_nodes.features.astype(dummy_mesh_node_features.dtype),
                batch_size)
        ],
                                 axis=-1))
```

the dummy inputs are being prepended, so I think you'd want to inspect the **last 3 rows** of **the weight matrix**.

The bias vector is being added to the projected feature vector in the latent space so I'm not sure you can truncate its inspection in any way.

Hope this helps.

-- Andrew

**Comment by divijkhaitan** [comment]
Right. Since I want just the 3 actual input coordinates to be considered, I can look at the entire bias vector and the last 3 **columns** of the weight matrix. Thanks for the response.

**Comment by andrewlkd** [comment]
No worries.

Just to clarify, as I think you have found, the weight matrix of the linear layer has shape (474, 512). 

I believe it's the last 3 components of the first dimension that you want to look at (what I would refer to as the rows, but indeed this can sometimes be confusing with notation based on how you define the matrix of the linear layer).
